pkgs/applications/editors/jupyter/lib.nix

```diff
@@ -0,0 +1,81 @@
+{ jupyter-kernel, python3, lib, ...}:
+{
+  /* Generates a kernel definition from a haskell environment
+
+      Example:
+        let ghcEnv = pkgs.haskellPackages.ghcWithPackages (p: [ p.aeson ]);
+        in mkKernelFromHaskellEnv ghcEnv;
+
+      Type:
+        mkKernelFromHaskellEnv :: Derivation -> AttrSet
+  */
+  mkKernelFromHaskellEnv = ghcEnv:
```

Comment:
I'd expect to have 1 file per kernel. And perhaps not constructing raw Jupyter kernelspecs like this, because then we have to worry about making sure they're all valid. IIRC there is currently some machinery in Nixpkgs for constructing these with some type safety.